src/mjolnir/bssbuilder.cc

```diff
@@ -0,0 +1,476 @@
+  std::list<PointLL> endshape = {};
+  PointLL bss_ll = {};
+
+  uint32_t start_to_bss_edge_idx = -1;
+  uint32_t end_to_bss_edge_idx = -1;
+
+  OSMConnectionEdge(GraphId startnode,
+                    GraphId endnode,
+                    uint64_t wayid,
+                    std::vector<std::string> names,
+                    std::list<PointLL> startshape,
+                    std::list<PointLL> endshape,
+                    const PointLL& bss_ll)
+      : startnode(startnode), endnode(endnode), wayid(wayid), names(std::move(names)),
+        startshape(std::move(startshape)), endshape(std::move(endshape)), bss_ll(bss_ll) {
+  }
+
+  // operator < for sorting
+  bool operator<(const OSMConnectionEdge& other) const {
+    if (startnode.tileid() == startnode.tileid()) {
+      return startnode.id() < other.startnode.id();
+    } else {
+      return endnode.id() < other.endnode.id();
+    }
+  }
+};
+
+template <typename T> struct Finally {
+  T t;
+  explicit Finally(T t) : t(t){};
+  Finally() = delete;
+  Finally(Finally&& f) = default;
+  Finally(const Finally&) = delete;
+  Finally& operator=(const Finally&) = delete;
+  Finally& operator=(Finally&&) = delete;
+  ~Finally() {
+    t();
+  };
+};
+
+template <typename T> Finally<T> make_finally(T t) {
+  return Finally<T>{t};
+};
+
+std::vector<OSMConnectionEdge> project(const GraphTile& local_tile,
+                                       const std::vector<OSMNode>& osm_bss) {
+
+  std::vector<OSMConnectionEdge> res;
+  auto local_level = TileHierarchy::levels().rbegin()->first;
+  // In this loop, we try to find the way on which to project the bss node by iterating all nodes in
+  // its corresponding tile... Not a good idea in term of performance... any better idea???
+  for (const auto& bss : osm_bss) {
+    auto latlng = bss.latlng();
+    const auto bss_ll = PointLL{latlng.first, latlng.second};
+    float mindist = 10000000.0f;
+    uint32_t edgelength = 0;
+    GraphId startnode, endnode;
+    std::vector<PointLL> closest_shape;
+    std::tuple<PointLL, float, int> closest;
+    std::vector<std::string> names;
+    uint32_t way_id = 0;
+    const DirectedEdge* best_directededge = nullptr;
+
+    // Loop over all nodes in the tile to find the nearest edge
+    for (uint32_t i = 0; i < local_tile.header()->nodecount(); ++i) {
+      const NodeInfo* node = local_tile.node(i);
+      for (uint32_t j = 0; j < node->edge_count(); ++j) {
+        const DirectedEdge* directededge = local_tile.directededge(node->edge_index() + j);
+        auto edgeinfo = local_tile.edgeinfo(directededge->edgeinfo_offset());
+
+        if (directededge->use() == Use::kTransitConnection ||
+            directededge->use() == Use::kEgressConnection ||
+            directededge->use() == Use::kPlatformConnection) {
+          continue;
+        }
+        if (!(directededge->forwardaccess() & kBicycleAccess) ||
+            !(directededge->forwardaccess() & kPedestrianAccess) || directededge->is_shortcut()) {
```

Comment:
Wouldn't you always want to connect to an edge with bicycle access? If you connect to a pedestrian only edge (no bike access) then once you leave the bike share station you wouldn't be able to continue any further on the connecting directed edge?

Reply:
OK, How about just ignoring edges that are not accessible neither by foot nor by bicycle?